### ISSUE TITLE
in Home Component,  can't use <>: Fragment.

### ISSUE THREAD

**Issue by 75asa** [yhatt/jsx-slack#266]
I always use jsx-slack for this kind of situation.
It is very helpful because I can write Block Kit very sensibly.
Thank you very much.

On topic, I tried to use Fragment when describing a component in the Home Block this time, but I couldn't describe Fragment using the <> notation that I always use.

After a while, I wondered if Fragment would work, so I modified the code and the jsx-slack transpile went through.

Is this a specification?
I have confirmed that the original <> works for <Blocks>.

If you know anything about this, I would like to know.

Thanx.

**Comment by yhatt** [comment]
The parent container block do _never_ restrict using `<>`. I guess the cause of not working `<>` is in your JSX transpile setting or your source code, not in jsx-slack. How do you have set up JSX transpiler?

#### If you're using TypeScript

* When `react-jsx`/`react-jsxdev` mode is selected in `jsx` compiler option of `tsconfig.json`:
  - :arrow_right: Check whether `jsx-slack` has defined in either of `jsxImportSource` compiler option or `/** @jsxImportSource **/` comment pragma.
* When `react` mode is selected in `tsconfig.json` (classic way):
  - :arrow_right: Check whether `JSXSlack.Fragment` has defined in either of `jsxFragmentFactory` compiler option or `/** @jsxFrag **/` comment pragma. [It requires TypeScript v4 and newer](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#custom-jsx-factories), and needs to import `JSXSlack` at the source code like below.<br /><br />
    ```typescript
    /** @jsx JSXSlack **/
    /** @jsxFrag JSXSlack.Fragment **/
    import { JSXSlack } from 'jsx-slack' /* <- Required to use JSXSlack.Fragment */
    ```

#### If you're using `jsxslack` template literal

You don't have to use fragments. It supports multiple elements in the top level.

**Comment by 75asa** [comment]
Thanx 4 your rapid reply @yhatt !!

This is my source code

```tsx
/** @jsx JSXSlack **/
/** @jsxFrag JSXSlack.Fragment **/
import { Divider, Fragment, Header, Image } from 'jsx-slack';
import { JSXSlack } from 'jsx-slack'; /* <- Required to use JSXSlack.Fragment */

export const HomeHeader = () => {
  return (
    <>
      <Image src="https://source.unsplash.com/random/960x240?home" alt="home" />
      <Header>SlackApp</Header>
      <Divider />
    </>
  );
};
```

And this is tsconfig.json

```json
{
  "compilerOptions": {
    "module": "commonjs",
    "jsx": "react-jsx",
    "jsxImportSource": "jsx-slack",
    "jsxFragmentFactory": "jsxFragmentFactory",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "lib": ["es2017", "dom"],
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true,
    "strictNullChecks": false,
    "noImplicitAny": false,
    "strictBindCallApply": false,
    "forceConsistentCasingInFileNames": false,
    "noFallthroughCasesInSwitch": false
  }
}
```

I'll take your advice.
I changed jsxFragmentFactory to "jsxFragmentFactory".
However, I am getting the following error
Is tsconfig.json wrong? Thanx.

```
[0:00:50] File change detected. Starting incremental compilation...

error TS5052: Option 'jsxFragmentFactory' cannot be specified without specifying option 'jsxFactory'.

error TS5089: Option 'jsxFragmentFactory' cannot be specified when option 'jsx' is 'react-jsx'.

[0:00:50] Found 2 errors. Watching for file changes.
```

**Comment by yhatt** [comment]
It looks like you are using `react-jsx` mode. In this case, `@jsx`, `@jsxFrag` comment pragma and `jsxFragmentFactory` option cannot use because they are only for `react` mode. Just forget previous advices 🤷

Following codes are minimum example to work `<>` syntax in `react-jsx` mode. Empty import for `jsx-slack` may resolve your problem.

```jsonc
// tsconfig.json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "jsx-slack"
  }
}
```

```tsx
// May require empty import at least, to make TypeScript compiler recognize jsx-slack.
import {} from 'jsx-slack'

export const Example = () => (
  <>
    <p>Test</p>
    <p>Hello!</p>
  </>
)
```

**Comment by yhatt** [comment]
UPDATE: Some of transpilers might have been updated to depend on `Fragment` that is not exported by jsx-slack JSX runtime. I've opened new issue in #267.

**Comment by yhatt** [comment]
Lacked `Fragment` export in JSX runtime (`jsx-slack/jsx-runtime`) has been fixed in the latest update [jsx-slack v4.6.1](https://github.com/yhatt/jsx-slack). TypeScript should resolve `Fragment` from JSX runtime automatically when you tried to use `<>` in your TSX code.
